commit fa834c27d783c290856d9d24fbc4a29acb8e4456
Author: Daniel Drake <dsd@laptop.org>
Date:   Fri Jun 1 15:45:04 2012 -0600

    power: update fallback status icon on icon state change
    
    When the icon was being changed from one icon to another, the fallback
    icon was not being updated.
    
    This meant that unplugging or re-plugging AC power on OLPC laptops
    was not updating the icon (even though the tooltip updated just fine).
    
    Add a missing call to update the GtkStatusIcon when the internal GIcon
    changes.
    
    Fixes http://dev.laptop.org/ticket/11878
    
    https://bugzilla.gnome.org/show_bug.cgi?id=677313

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 1d60723..abec33f 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -711,6 +711,10 @@ engine_recalculate_state_icon (GsdPowerManager *manager)
 
         /* icon before, now different */
         if (!g_icon_equal (manager->priv->previous_icon, icon)) {
+
+                /* set fallback icon */
+                gtk_status_icon_set_from_gicon (manager->priv->status_icon, icon);
+
                 g_object_unref (manager->priv->previous_icon);
                 manager->priv->previous_icon = icon;
                 return TRUE;
